podcast/2.15.md

```diff
@@ -0,0 +1,1299 @@
+
+I think it was Akamai. Most likely Akamai because that name does ring a bell. So we stored the video data and Akamai would spread it around the world and take care of that part. But it was metadata. Metadata was growing too fast. We couldn't keep up.
+
+**Beyang Liu:**
+
+What does metadata encompass? Is that comments and descriptions and titles?
+
+**Sugu Sougoumarane:**
+
+Comments, videos, user information. Apparently, the insert rate in the video was so high that it could not keep up with the inserts. Because we could not insert fast enough.
+
+**Beyang Liu:**
+
+There were so many comments that people were living on videos, or?
+
+**Sugu Sougoumarane:**
+
+So many videos being created.
+
+**Beyang Liu:**
+
+Oh, wow. Okay. So just the number of videos being created.
+
+**Sugu Sougoumarane:**
+
+So we're talking hard disks, right? This is 2005, 2006.
+
+**Beyang Liu:**
+
+So, spinning discs.
+
+**Sugu Sougoumarane:**
+
+Spinning discs. You can only insert so fast.
+
+**Beyang Liu:**
+
+Wow. Okay, so it was too big for even the largest single box money could buy. Too big for one MySQL instance. And so you started working on sharding it. At what point did sharded MySQL become Vitess? Was there a single point where you decided to give a name to it? Or was it even the same project?
+
+**Sugu Sougoumarane:**
+
+So, we had sharded the MySQL database. We had, I believe, eight shards if I remember correctly. Because eight shards should be more than enough, right?
+
+**Beyang Liu:**
+
+Eight shards should be enough for anyone.
+
+**Sugu Sougoumarane:**
+
+I mean, we went so far with one shard. With eight shards, we can scale 8X. Here we are all set for the rest of our lives, is what we thought. And what was happening was that, at eight shards, the shards themselves were okay. We had not reached their capacity. But what had happened was that managing those eight shards became a huge pain for a number of reasons. And there were very frequent outages. A couple of outages a day kind of thing.
+
+**Beyang Liu:**
+
+Because one shard would go down?
+
+**Sugu Sougoumarane:**
+
+There was no peace in our organization. There was no peace. Because there was always an outage going on, and every time it was the database. So it was almost like, "Oh, there's an outage. Let's start looking at the database because that's where the problem is." And most of the time, that was the case. And so the thing that my co-founder, co-creator, and I did–actually, most of that work was done by him. What he did was, he took himself out and wrote a spreadsheet. There was no postmortem process those days.
+
+**Beyang Liu:**
+
+No Five Why's.
+
+**Sugu Sougoumarane:**
+
+Yeah, none of that stuff. But he kind of went through that. He made a spreadsheet of every failure that we had ever faced. Every one of those outages. What caused it and what can we do to prevent it? And so he wrote–iit was a huge spreadsheet. I may even be able to dig it up again. I should it somewhere.
```

Comment:
```suggestion
Yeah, none of that stuff. But he kind of went through that. He made a spreadsheet of every failure that we had ever faced. Every one of those outages. What caused it and what can we do to prevent it? And so he wrote–it was a huge spreadsheet. I may even be able to dig it up again. I should it somewhere.
```